amgut/lib/data_access/ag_data_access.py

```diff
@@ -323,7 +340,10 @@ def deleteAGParticipantSurvey(self, ag_login_id, participant_name):
                 self.deleteSample(info['barcode'], ag_login_id)
 
             # Delete last due to foreign keys
-            sql = "DELETE FROM ag_kit_barcodes WHERE survey_id IN %s"
+            sql = "DELETE FROM ag.ag_kit_barcodes WHERE barcode IN %s"
```

Comment:
hmmm ... the table source_barcode_surveys must be able to store multiple barcodes for the same survey_id. Does a FK restriction on DELETE work if the column is NOT unique?